### PR TITLE
[tests] Ensure aggregation integration test uses remote aggregators

### PR DESCRIPTION
**Pull request by nbroyles**
**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/integration/aggregator/aggregator.go [hunk 1]
```diff
@@ -69,6 +79,43 @@ downsample:
         storagePolicies:
           - resolution: 5s
             retention: 6h
+  remoteAggregator:
```
Comment (nbroyles) [inline src/integration/aggregator/aggregator.go:82]: This is what signals to the coordinator to not use the embedded aggregator.

File: src/integration/aggregator/aggregator.go [hunk 2]
```diff
@@ -128,9 +175,9 @@ func testAggregatedGraphiteMetric(t *testing.T, m3 resources.M3Resources) {
 		}
 	}()
 
-	require.NoError(t, resources.Retry(func() error {
+	require.NoError(t, resources.RetryWithMaxTime(func() error {
 		return verifyGraphiteQuery(m3, carbonTarget, expectedCarbonMean)
-	}))
+	}, 2*time.Minute))
```
Comment (nbroyles) [inline src/integration/aggregator/aggregator.go:180]: Aggregation takes a slight bit longer now that we're using separate aggregators.

File: src/integration/aggregator/aggregator.go [hunk 3]
```diff
@@ -166,7 +213,7 @@ func filterNull(datapoints []resources.Datapoint) []resources.Datapoint {
 func testRollupRule(t *testing.T, m3 resources.M3Resources) {
 	var (
 		numDatapoints = 5
-		resolutionSec = 10
+		resolutionSec = 5
```
Comment (nbroyles) [inline src/integration/aggregator/aggregator.go:216]: This was intended to match the namespace resolution.

File: src/integration/resources/inprocess/cluster.go [hunk 4]
```diff
@@ -233,6 +238,66 @@ func NewClusterFromSpecification(
...
+			return errors.New("remote aggregation requires at least one DB node" +
+				" running an embedded coordinator for placement and topic configuration")
+		}
+
+		embedded, err := NewEmbeddedCoordinator(nodes[0].(*DBNode))
+		if err != nil {
+			return nil
+		}
+
+		coordAPI = embedded
```
Comment (fishie9) [inline src/integration/resources/inprocess/cluster.go:276]: If the embedded coordinator is used here, are we assuming the coordinator in the cluster specification is always using m3db for the kv store?

File: src/integration/aggregator/aggregator.go [hunk 5]
```diff
@@ -39,11 +39,21 @@ const (
 	// TestAggregatorDBNodeConfig is the test config for the dbnode.
 	TestAggregatorDBNodeConfig = `
 db: {}
+coordinator: {}
 `
 
 	// TestAggregatorCoordinatorConfig is the test config for the coordinator.
 	TestAggregatorCoordinatorConfig = `
 listenAddress: 0.0.0.0:7202
+metrics:
```
[the remark this answers is not in the document]
Comment (nbroyles) [reply src/integration/aggregator/aggregator.go:48]: There's a collision between the embedded coordinator and standalone coordinator now, so we have to modify one.

File: src/integration/resources/inprocess/cluster.go (hunk 4 above)
Comment (nbroyles) [reply src/integration/resources/inprocess/cluster.go:276]: Not exactly. If the coordinator in the cluster specification wants to use a different KV store, then they'll just need to update the kv store used in the embedded coordinators config. Practically though, I don't see anyone changing to configs to modify which kvstore to use (even though we support it)

File: src/integration/resources/inprocess/cluster.go [hunk 6]
```diff
@@ -233,6 +238,66 @@ func NewClusterFromSpecification(
...
+	} else {
+		// TODO(nate): Remove this in a follow up. If we're not doing remote aggregation
+		// we should not be starting aggs which is what requires the coordinator to get started.
+		// Once we've refactored existing tests that have aggs w/o remote aggregation enabled,
```
Comment (wesleyk) [inline src/integration/resources/inprocess/cluster.go:280]: "aggs w/o" -> "coords w/o"?

**Review by wesleyk — APPROVED** [review]

File: src/integration/resources/inprocess/coordinator.go [hunk 7]
```diff
@@ -207,7 +214,15 @@ func NewEmbeddedCoordinator(d *DBNode) (resources.Coordinator, error) {
 	}, nil
 }
 
-func (c *Coordinator) start() {
+// Start is the start method for the coordinator.
+//nolint:dupl
+func (c *Coordinator) Start() {
```
Comment (wesleyk) [inline src/integration/resources/inprocess/coordinator.go:219]: when would this be triggered twice?

**Pull request merged**